doc/languages-frameworks/rust.md

```diff
@@ -261,6 +264,46 @@ general. A number of other parameters can be overridden:
   };
   ```
 
+### Setting Up `nix-env`
+Oftentimes you want to develop code using `nix-env`. Using the example above,
+we want to do the following:
+
+- Have access to `cargo` and `rustc`
+- Have the `openssl` library available to the rust crate through
+  it's _normal_ compilation mechanism (`pkg-config`).
+- Examples of additional library paths is also given.
+
+
+`default.nix`:
+```
+with import <nixpkgs> {};
+((import ./hello.nix).hello {}).override {
+  crateOverrides = defaultCrateOverrides // {
+    hello = attrs: {
+        buildInputs = [
+        # Some example external dependencies
+        openssl xorg.libX11 xorg.libXtst
+        # For `nix-shell`
+        rustc cargo openssl pkgconfig
+      ];
+
+      # Set Relevant ENV Variables
+      LIBRARY_PATH = "${xorg.libX11}/lib:${xorg.libXtst}/lib";
```

Comment:
This should not be necessary.

Reply:
wow, you're right -- it's not. I just did a cargo clean+build without it and it worked